### PR TITLE
 "Feature/processing data"

### PR DESCRIPTION
**Pull request by peterdgreat**
- Created model classes for all entities as shown in the ERD diagram.
- Set up associations between models.

### CODE REVIEW

File: app/models/comment.rb [hunk 1]
```diff
@@ -1,10 +0,0 @@
-class Comment < ApplicationRecord
-  belongs_to :author, class_name: 'User'
-  belongs_to :post
-  after_save :comment_update_counter
-
-  def comment_update_counter
-    post = Post.find(post_id)
-    post.update_attribute(:comments_count, post.comments_count + 1)
-  end
-end
```
Comment (Dipeshtwis) [inline app/models/comment.rb]: - [ ] I think you have deleted all the association and model work that was needed in this milestone and your application. Please fix this.

File: app/controllers/application_controller.rb [hunk 2]
```diff
@@ -1,5 +1,2 @@
 class ApplicationController < ActionController::Base
-  def current_user
-    User.first
-  end
```
Comment (Dipeshtwis) [inline app/controllers/application_controller.rb:4]: - [ ] I think you have deleted all the association and model work that was needed in this milestone and your application. Please fix this.

File: app/models/post.rb [hunk 3]
```diff
@@ -1,18 +0,0 @@
...
-
-  def post_update_counter
-    user = User.find(author_id)
-    user.update_attribute(:posts_count, user.posts_count + 1)
-  end
-
-  def most_recent_comments
-    Comment.where(post_id: id).order(created_at: :desc).limit(5)
-  end
-end
```
Comment (Dipeshtwis) [inline app/models/post.rb]: - [ ] I think you have deleted all the association and model work that was needed in this milestone and your application. Please fix this.

File: app/models/user.rb [hunk 4]
```diff
@@ -1,9 +0,0 @@
-class User < ApplicationRecord
-  has_many :posts, foreign_key: :author_id
-  has_many :comments, foreign_key: :author_id
-  has_many :likes, foreign_key: :author_id
-
-  def most_recent_3_posts
-    Post.where(author_id: id).order(created_at: :desc).limit(3)
-  end
-end
```
Comment (Dipeshtwis) [inline app/models/user.rb]: - [ ] I think you have deleted all the association and model work that was needed in this milestone and your application. Please fix this.

**Review by Dipeshtwis — CHANGES_REQUESTED** [review]
Hi @peterdgreat, 

:+1: This is good to make it this far, your project is really nice, congratulations on the good work you have done so far.

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!


 ## Required Changes ♻️

**_Check the comments under the review_**

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

- N.A.

Cheers and Happy coding!👏👏👏
Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

**_Dipesh Kumar_**


------

_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project(check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by peterdgreat** [comment]
Ohh, I opened the wrong branch for review

**Review by Emmyn5600 — APPROVED** [review]
Hi @peterdgreat 

Woo ow 👏, Your project is complete!  There is nothing else to say other than... it's time to merge it. 👏

Congratulations! 🎉 

##  Optional suggestion

- N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**Pull request merged**